src/utils.rs

```diff
@@ -48,3 +49,9 @@ pub fn hex_hash<T: Hash>(t: &T) -> String {
     t.hash(&mut s);
     format!("{:#x}", s.finish())
 }
+
+//TODO verify we are only touching 'offset' and 'limit'
```

Comment:
I would probably call is extract query string as we can also use it for other cases then the next. Also I don't think we need to check that it is only offset and limit, if backend changes the pagination it would still work ... Probably